authors/apps/authentication/views.py

```diff
@@ -146,4 +152,57 @@ def get(self, request):
         favorite_articles = user.favorite.all()
         serializer = self.serializer_class(favorite_articles, many=True)
         return Response(serializer.data, status=status.HTTP_200_OK)
-  
\ No newline at end of file
+  
+class PasswordResetView(GenericAPIView):
+    """ Allow users to receive a password reset link intheir emails"""
+    
+    permission_classes = (AllowAny,)
+    renderer_classes = (UserJSONRenderer,)
+    serializer_class = ResetPasswordSerializer
+
+    def post(self, request):
+        user_email = request.data.get('email')
+        user = User.objects.filter(email=user_email).first()
+        if user:
+            token = user._generate_jwt_token()
+            url = get_current_site(request).domain + reverse('password-change',kwargs={'token':token})
+            send_email = EmailMessage(
+                subject='Authors Haven:Password Reset',
+                body='Click here to reset your password http://{}'.format(url),
+                from_email= settings.EMAIL_HOST_USER,
+                to=[user.email]
+                )
+            send_email.send(fail_silently=False)
+            return Response({'message':'Password reset link has been sent to your Email'},
+            status=status.HTTP_200_OK)
+            
+        return Response({'message':'Email does not exist'},
+        status=status.HTTP_400_BAD_REQUEST)
+
+class PasswordChangeView(GenericAPIView):
+
+    permission_classes = (AllowAny,)
+    renderer_classes = (UserJSONRenderer,)
+    serializer_class = ConfirmPasswordSerializer
+
+    def put(self, request, *args, **kwargs):
+        token = kwargs.pop('token')
+        user = jwt.decode(token,SECRET_KEY)
+
+        user_data = User.objects.get(pk=user['id'])
+        data = request.data.get('user', {})
+    
+        if data.get('password') == data.get('confirm_password'):
+            serializer = self.serializer_class(user_data, data=data)
+            serializer.is_valid(raise_exception=True)
```

Comment:
@dorothyas Please put the serializer logic lines `196` and `197` before the `if` statement so that validation if done before the `if` statement